app/controllers/ManageClientController.scala

```diff
@@ -61,59 +61,33 @@ class ManageClientController @Inject()(
   def showAllClients: Action[AnyContent] = Action.async { implicit request =>
     isAuthorisedAgent { arn =>
       isOptedIn(arn) { _ =>
-        withSessionItem[Seq[DisplayClient]](FILTERED_CLIENTS) { maybeFilteredResult =>
-          if (maybeFilteredResult.isDefined)
-            Ok(
-              manage_clients_list(
-                maybeFilteredResult,
-                AddClientsToGroupForm.form()
-              )
-            ).toFuture
-          else
-            groupService.getClients(arn).flatMap { maybeClients =>
+        groupService.getClients(arn).flatMap { maybeClients =>
+          val searchFilter: SearchFilter = SearchAndFilterForm.form().bindFromRequest().get
+          searchFilter.submit.fold(
+            //no filter/clear was applied
+            Ok(manage_clients_list(
+                maybeClients,
+                SearchAndFilterForm.form()
+            )).toFuture
+          )({
+            //clear/filter buttons pressed
+            case "clear" =>
```

Comment:
we have the ButtonSelect object to avoid using Strings?

Reply:
ButtonSelect has continue, so it keeps throwing warnings that "'ButtonSelect' is never continue"

maybe we could make another object/case for just filter & clear?

Reply:
yes let's do that when we have time